src/handlers/autolabel.rs

```diff
@@ -69,11 +85,18 @@ pub(super) async fn parse_input(
                         name: label.to_owned(),
                     });
                 }
-                if cfg.new_pr && event.action == IssuesAction::Opened {
-                    autolabels.push(Label {
-                        name: label.to_owned(),
-                    });
-                }
+            }
+
+            // Newly opened non-draft PRs or PRs that were opened as draft and that were marked as
+            // "ready for review" for the first time are treated as "new PRs".
+            let treat_as_new_pr = (event.action == IssuesAction::Opened && !event.issue.draft)
+                || (event.action == IssuesAction::ReadyForReview
+                    && !state.data.new_pr_labels_applied);
```

Comment:
The logic here is a bit dense. Just for improving a bit the readability, could it be split into something in this direction:
```suggestion
            // Treat as new PR:
			// - 1) New PRs, exclude drafts
			// - 2) PRs opened as draft and then tagged as "ready for review"
            let is_new_draft_pr = (event.action == IssuesAction::Opened && !event.issue.draft));
            let is_ready_for_review = (event.action == IssuesAction::ReadyForReview
                    && !state.data.new_pr_labels_applied);
			let treat_as_new_pr = !is_new_draft_pr || is_ready_for_review;
```

(unsure about ny rewording, I can't test right now)